### PR TITLE
add TraceApi impl placeholder, todo for functions implemention

### PR DESCRIPTION
**Pull request by CryptoTurboFish**
For #982

### CODE REVIEW

File: crates/net/rpc/src/trace.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    trace::{filter::TraceFilter, parity::*},
+    CallRequest, Index,
+};
+use std::collections::HashSet;
+
+/// `trace` API implementation.
+///
+/// This type provides the functionality for handling `trace` related requests.
+pub struct TraceApi {
+    network: NetworkHandle,
```
Comment (mattsse) [inline crates/net/rpc/src/trace.rs:16]: we'll phase out `NetworkHandle` entirely, see #979 

so let's rm that here, need to figure out first what we actually need.

**Review by mattsse — CHANGES_REQUESTED** [review]
ty, 
smol nit

File: crates/net/rpc/src/trace.rs (hunk 1 above)
Comment (CryptoTurboFish) [reply crates/net/rpc/src/trace.rs:16]: i see, will update

**Review by mattsse — APPROVED** [review]
lgtm

**Review by rkrasiuk — APPROVED** [review]

**Pull request merged**